app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :likes
+  has_many :comments
+
+  def update_posts_count
+    author.update(posts_counter: author.posts.size)
+  end
+
```

Comment:
- [ ] [OPTIONAL] We can make use of some model callback functions such as __after_save__.

```ruby
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter) # using the increment function will increment the posts_counter without calling on size function.
  end
```

Reply:
@elisha2kyakpo1 thank you for these suggestions, I was unaware of the callback functions.